### PR TITLE
[clang-tidy] use auto

### PR DESCRIPTION
**Pull request by neheb**
Found with modernize-use-auto

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/stream.cpp [hunk 1]
```diff
@@ -151,7 +151,7 @@ inline UtfIntroCharType IntroCharTypeOf(std::istream::int_type ch) {
 
 inline char Utf8Adjust(unsigned long ch, unsigned char lead_bits,
                        unsigned char rshift) {
-  const unsigned char header =
+  const auto header =
```
Comment (jbeder) [inline src/stream.cpp:154]: Leave this one as-is; I don’t think auto helps here.

Comment (neheb) [reply src/stream.cpp:154]: This was automatically done. The idea is to avoid duplicating the type name since there's a static_cast.

Comment (jbeder) [reply src/stream.cpp:154]: I know, but I don’t agree with the philosophy. I prefer auto only when (1) the type is obvious and (2) the type name is unwieldy to write. This matches (1) but not (2).

Comment (neheb) [reply src/stream.cpp]: OK. Added nolint.

Comment (jbeder) [reply src/stream.cpp]: Could you keep off the NOLINT? I don’t think that adds anything either, since there are so many different linters we could use.

Comment (neheb) [reply src/stream.cpp]: OK.

**Pull request merged**